File: smallpict-java/src/main/java/com/smallpict/models/QuotaResponse.java
package com.smallpict.models;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class QuotaResponse {
    private final String plan;
    @JsonProperty("bytes_used")
    private final long bytesUsed;
    @JsonProperty("quota_limit")
    private final long quotaLimit;
    @JsonProperty("quota_percentage")
    private final double quotaPercentage;
    @JsonProperty("cdn_egress_used_bytes")
    private final Long cdnEgressUsedBytes;
    @JsonProperty("cdn_egress_quota_bytes")
    private final Long cdnEgressQuotaBytes;
    @JsonProperty("active_keys_count")
    private final Integer activeKeysCount;
    @JsonProperty("active_sites_count")
    private final Integer activeSitesCount;

    @JsonCreator
    public QuotaResponse(
            @JsonProperty("plan") String plan,
            @JsonProperty("bytes_used") Long bytesUsed,
            @JsonProperty("quota_limit") Long quotaLimit,
            @JsonProperty("quota_percentage") Double quotaPercentage,
            @JsonProperty("cdn_egress_used_bytes") Long cdnEgressUsedBytes,
            @JsonProperty("cdn_egress_quota_bytes") Long cdnEgressQuotaBytes,
            @JsonProperty("active_keys_count") Integer activeKeysCount,
            @JsonProperty("active_sites_count") Integer activeSitesCount) {
        this.plan = plan != null ? plan : "free";
        this.bytesUsed = bytesUsed != null ? bytesUsed : 0L;
        this.quotaLimit = quotaLimit != null ? quotaLimit : 0L;
        if (quotaPercentage != null) {
            this.quotaPercentage = Math.round(quotaPercentage * 100.0) / 100.0;
        } else if (this.quotaLimit > 0) {
            this.quotaPercentage = Math.round(((double) this.bytesUsed / this.quotaLimit * 100.0) * 100.0) / 100.0;
        } else {
            this.quotaPercentage = 0.0;
        }
        this.cdnEgressUsedBytes = cdnEgressUsedBytes;
        this.cdnEgressQuotaBytes = cdnEgressQuotaBytes;
        this.activeKeysCount = activeKeysCount;
        this.activeSitesCount = activeSitesCount;
    }

    public String getPlan() {
        return plan;
    }

    public long getBytesUsed() {
        return bytesUsed;
    }

    public long getQuotaLimit() {
        return quotaLimit;
    }

    public double getQuotaPercentage() {
        return quotaPercentage;
    }

    public Long getCdnEgressUsedBytes() {
        return cdnEgressUsedBytes;
    }

    public Long getCdnEgressQuotaBytes() {
        return cdnEgressQuotaBytes;
    }

    public Integer getActiveKeysCount() {
        return activeKeysCount;
    }

    public Integer getActiveSitesCount() {
        return activeSitesCount;
    }
}

File: smallpict-php/src/Models/QuotaResponse.php
<?php

declare(strict_types=1);

namespace SmallPict\Models;

class QuotaResponse
{
    private string $plan;
    private int $bytesUsed;
    private int $quotaLimit;
    private float $quotaPercentage;
    private ?int $cdnEgressUsedBytes;
    private ?int $cdnEgressQuotaBytes;
    private ?int $activeKeysCount;
    private ?int $activeSitesCount;

    public function __construct(
        string $plan,
        int $bytesUsed,
        int $quotaLimit,
        float $quotaPercentage,
        ?int $cdnEgressUsedBytes = null,
        ?int $cdnEgressQuotaBytes = null,
        ?int $activeKeysCount = null,
        ?int $activeSitesCount = null
    ) {
        $this->plan = $plan;
        $this->bytesUsed = $bytesUsed;
        $this->quotaLimit = $quotaLimit;
        $this->quotaPercentage = $quotaPercentage;
        $this->cdnEgressUsedBytes = $cdnEgressUsedBytes;
        $this->cdnEgressQuotaBytes = $cdnEgressQuotaBytes;
        $this->activeKeysCount = $activeKeysCount;
        $this->activeSitesCount = $activeSitesCount;
    }

    public function getPlan(): string
    {
        return $this->plan;
    }

    public function getBytesUsed(): int
    {
        return $this->bytesUsed;
    }

    public function getQuotaLimit(): int
    {
        return $this->quotaLimit;
    }

    public function getQuotaPercentage(): float
    {
        return $this->quotaPercentage;
    }

    public function getCdnEgressUsedBytes(): ?int
    {
        return $this->cdnEgressUsedBytes;
    }

    public function getCdnEgressQuotaBytes(): ?int
    {
        return $this->cdnEgressQuotaBytes;
    }

    public function getActiveKeysCount(): ?int
    {
        return $this->activeKeysCount;
    }

    public function getActiveSitesCount(): ?int
    {
        return $this->activeSitesCount;
    }
}
